pkg/cmd/server/start.go

```diff
@@ -261,18 +261,18 @@ func start(cfg *config, args []string) error {
 		}
 
 		osmaster := &origin.MasterConfig{
-			TLS:                   cfg.BindAddr.URL.Scheme == "https",
-			MasterBindAddr:        cfg.BindAddr.URL.Host,
-			MasterAddr:            cfg.MasterAddr.URL.String(),
-			MasterPublicAddr:      masterPublicAddr.URL.String(),
-			AssetBindAddr:         assetBindAddr,
-			AssetPublicAddr:       assetPublicAddr.String(),
-			KubernetesAddr:        cfg.KubernetesAddr.URL.String(),
-			KubernetesPublicAddr:  k8sPublicAddr.URL.String(),
-			EtcdHelper:            etcdHelper,
-			RequireAuthentication: cfg.RequireAuthentication,
-			Authorizer:            apiserver.NewAlwaysAllowAuthorizer(),
-			AdmissionControl:      admit.NewAlwaysAdmit(),
+			TLS:                          cfg.BindAddr.URL.Scheme == "https",
+			MasterBindAddr:               cfg.BindAddr.URL.Host,
+			MasterAddr:                   cfg.MasterAddr.URL.String(),
+			MasterPublicAddr:             masterPublicAddr.URL.String(),
+			AssetBindAddr:                assetBindAddr,
+			AssetPublicAddr:              assetPublicAddr.String(),
+			KubernetesAddr:               cfg.KubernetesAddr.URL.String(),
+			KubernetesPublicAddr:         k8sPublicAddr.URL.String(),
+			EtcdHelper:                   etcdHelper,
+			RequireAuthentication:        cfg.RequireAuthentication,
+			AdmissionControl:             admit.NewAlwaysAdmit(),
+			MasterAuthorizationNamespace: "master",
```

Comment:
> should we construct the authorizer outside the master and specify it in the config, similar to the way we pass in the authenticator and admission control objects?

In principle I agree.  I'd prefer a separate commit though, because it's actually going to pull a few unexpected objects out of master.go in order to share the same in-memory copy of the requestsToUsersMap with the authentication wrapper.  That may take a try or to two to make something not-ugly.